logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/conversation/JoinExistingMLSConversationUseCase.kt

```diff
@@ -107,29 +104,44 @@ class JoinExistingMLSConversationUseCaseImpl(
             }
 
     private suspend fun joinOrEstablishMLSGroup(conversation: Conversation): Either<CoreFailure, Unit> {
-        return if (conversation.protocol is Conversation.ProtocolInfo.MLSCapable) {
-            if (conversation.protocol.epoch == 0UL) {
-                if (conversation.type == Conversation.Type.SELF) {
+        val protocol = conversation.protocol
+        if (protocol !is Conversation.ProtocolInfo.MLSCapable) {
+            return Either.Right(Unit)
+        }
+        return if (protocol.epoch == 0UL) {
```

Comment:
can w combine these `if` and `when(conversation.type)` into one when
```
when {
   protocol.epoch != 0UL -> ...
   conversation.type == Conversation.Type.SELF -> ...
   conversation.type == Conversation.Type.ONE_ON_ONE -> ...
   else -> ...
}
```

just a thought, it's up to you :)

Reply:
Nice catch!